backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java

```diff
@@ -111,23 +137,39 @@ private WarehouseId findPickingWarehouseId(@NonNull final I_C_Order order)
 			return null;
 		}
 
-		final IBPartnerDAO bpartnersRepo = Services.get(IBPartnerDAO.class);
-		final I_C_BPartner bp = bpartnersRepo.getById(bpartnerId);
-		if (!bp.isCustomer())
-		{
-			return null;
-		}
-
-		final WarehouseId customerWarehouseId = WarehouseId.ofRepoIdOrNull(bp.getM_Warehouse_ID());
+		final WarehouseId customerWarehouseId = findWarehouseIdForSOTrx(bpartnerId, SOTrx.SALES);
 		if (customerWarehouseId != null && isPickingWarehouse(customerWarehouseId))
 		{
 			return customerWarehouseId;
 		}
 
-		// if order is a purchase order, return null
 		return null;
 	}
 
+	@Nullable
+	private WarehouseId findWarehouseIdForSOTrx(@NonNull final BPartnerId bpartnerId, @NonNull final SOTrx soTrx)
+	{
+		final IBPartnerDAO partnerDAO = Services.get(IBPartnerDAO.class);
+		final I_C_BPartner bp = partnerDAO.getById(bpartnerId);
+
+		if (soTrx.isSales())
+		{
+			if (!bp.isCustomer())
+			{
+				return null;
+			}
+			return WarehouseId.ofRepoIdOrNull(bp.getM_Warehouse_ID());
+		}
+
+		if (!bp.isVendor())
+		{
+			return null;
+		}
```

Comment:
i find this logic a bit hard to read
why not writing like
```java
if (soTrx.isSales()) && bp.isCustomer()) { return ... }
else if (soTrx.isPurchase() && bp.isVendor()) { return ... }
```

Reply:
Done